From 459692cfae48b0f6354fdf1c048297e264128c56 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 27 Jul 2021 00:03:36 +0800
Subject: [PATCH 08/12] dt-bindings: connector: add binding for possible
 altmodes for Type-C

As we already have some code to parse fwnode to populate alternate modes
(currently for x86 platform-code-generated fwnodes, not DT ones), add a
binding for altmodes that is close to it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../bindings/connector/usb-connector.yaml     | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/Documentation/devicetree/bindings/connector/usb-connector.yaml b/Documentation/devicetree/bindings/connector/usb-connector.yaml
index 92b49bc37939..1ae76c5bcf83 100644
--- a/Documentation/devicetree/bindings/connector/usb-connector.yaml
+++ b/Documentation/devicetree/bindings/connector/usb-connector.yaml
@@ -189,6 +189,34 @@ properties:
     required:
       - port@0
 
+  altmodes:
+    description: Optional subnode to describe what alternate modes are
+      to be supported.
+
+    properties:
+      "#address-cells":
+        const: 1
+
+      "#size-cells":
+        const: 0
+
+      "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@[0-9a-fA-F]+$":
+        type: object
+        properties:
+          reg:
+            maxItems: 1
+            description: The index of the alternate mode, starting at 1.
+
+          svid:
+            $ref: /schemas/types.yaml#/definitions/uint32
+            description: The SVID of the alternate mode.
+
+          vdo:
+            $ref: /schemas/types.yaml#/definitions/uint32
+            description: The VDO of the alternate mode.
+
+          additionalProperties: false
+
   new-source-frs-typec-current:
     description: Initial current capability of the new source when vSafe5V
       is applied during PD3.0 Fast Role Swap. "Table 6-14 Fixed Supply PDO - Sink"
-- 
2.30.2

